### PR TITLE
refactor: add more granular logging to cron job

### PR DESCRIPTION
**Pull request by JBR90**
## Description

Google drive delete cron job is throwing errors from Prisma, 

-  Update each record 1 at a time to identify [error](https://oak-national-academy.sentry.io/issues/11846058/?alert_rule_id=2299&alert_type=issue&environment=production&notification_uuid=accb24d2-7777-4df8-a3f3-f031453a52a0&project=4507089597300816&referrer=slack) with specific id's
- Allow job to continue to delete files when an id is not found in the database and report missing ids at the end with error

How to test: (comment out auth)
http://localhost:2525/api/cron-jobs/expired-exports

https://www.notion.so/oaknationalacademy/Google-Drive-file-deletion-cron-job-14426cc4e1b180b1bf88dd8d2c566ad1

## Checklist

- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: apps/nextjs/src/app/api/cron-jobs/expired-exports/route.ts [hunk 1]
```diff
@@ -15,47 +15,53 @@ requiredEnvVars.forEach((envVar) => {
...
-    }
-  } catch (error) {
-    log.error("Error updating expiredAt field in the database:", error);
-    throw error;
-  }
-}
+  for (const id of fileIds) {
+    try {
+      const record = await prisma.lessonExport.findFirst({
+        where: { gdriveFileId: id },
```
Comment (JBR90) [inline apps/nextjs/src/app/api/cron-jobs/expired-exports/route.ts:29]: Not unique value in database so find first to get lesson id

Comment (mantagen) [reply apps/nextjs/src/app/api/cron-jobs/expired-exports/route.ts:29]: Would we not want to get all rather than just the first one?

Comment (mantagen) [reply apps/nextjs/src/app/api/cron-jobs/expired-exports/route.ts:29]: > I could findAll and throw error if it finds more than one
Could we not findAll and updateMany?
Other option would be to add a unique constraint on that field

Comment (JBR90) [reply apps/nextjs/src/app/api/cron-jobs/expired-exports/route.ts:29]: The trouble with update many is that it only returns a count of the updated ids and not any info on failed ids

Comment (mantagen) [reply apps/nextjs/src/app/api/cron-jobs/expired-exports/route.ts:29]: Sure, fair enough. Let's put a unique constraint on it then, as we're not expecting multiple records per file! I'll put that in a separate PR

**Comment by JBR90** [comment]
> Do you have an idea of how much longer this approach will take to run? Thinking the function could time out -- but happy to trial it

Lets give it a go, i will at least give us some more info from the logs on ID's causing problem as if loops round.

**Pull request merged**